### ISSUE TITLE
hctreemap() fails to render with cellnames repeated across levels

### ISSUE THREAD

**Issue by alexpghayes** [jbkunst/highcharter#310]
Treemaps with cell names that are not unique across all levels silently fail to render:

```{r}
library(highcharter)
library(treemap)
library(magrittr)

tm <- data.frame(level_1 = c("Unknown", "A", "A", "B", "B"),
                 level_2 = c("Unknown", "C", "C", "D", "D"),
                 value = abs(rnorm(5))) %>%
  treemap(index = c("level_1", "level_2"),
          vSize = "value")

tm %>% hctreemap() # plot doesn't render
```

Only came up because missing values in my dataset were coded as "Unknown" and `d3treeR` supported this:

```{r}
# devtools::install_github("timelyportfolio/d3treeR")
library(d3treeR)

tm %>% d3tree2()
```

Would love to have this if it's easy to implement. If it's not, I'd go for a useful error message -- I've never made a pull request before but this seems like a good place to start.

**Comment by jbkunst** [comment]
related with #117

**Comment by alexpghayes** [comment]
think this is copy of #32.  have code to deal with this as well as #117 and #98 and hoping to make a PR in the next couple days.

like the `treemap2` function in /dev i'm currently working with a data frame rather than a treemap object -- you okay changing the hctreemap interface to something like 

`hctreemap(data, group_vars, size_var, color_var = NULL, ...)`?

not sure what to do with the `hctreemap(tm, ...)` interface though...

i can see people wanting to pass in `treemap` objects, but that turns out to be ~100 times slower than just manipulating the data frame with dplyr in my current experience, plus takes extra wrangling

**Comment by jbkunst** [comment]
Hi @alexpghayes 

What will you do in the PR? I'm asking to avoid duplicate efforts.

I'm okay changing the interface, but not sure how approach I will take to deprecate the actual `hctreemap`, for me I would replace the function, but surely is not the best approach.

Do you have any thoughts?

**Comment by alexpghayes** [comment]
PR will just replace `hctreemap` with new implementation along the lines of your `hctreemap2` except:

* uses dev version of `dplyr` (i.e. new quosure system in 0.5.0.9005+) rather than `data.tree`. hopefully 0.6.0 will come out soon.
* throws an error when index columns have non-empty intersection

Maybe keep the current `hctreemap`, call the new function `hctreemap2` and give a deprecation warning? I'd understand wanting to keep an interface that allows passing in a `treemap::treemap` object, but not sure if you wanna have two distinct interfaces.

Also I haven't written code to deal with treemap objects -- they take a noticable amount of time to create for large treemaps (500+). Not that treemaps of that size are necessarily a good idea, but...

Just replacing the function would break things, but the fix would be super easy. My version of `hctreemap2` needs the exact same data that goes into `treemap::treemap`. The biggest difference I think is controlling the color palette--what I've got uses a colorAxis. I'm not sure how to give more customization control.

**Comment by jbkunst** [comment]
Fixed in #327 by @alexpghayes  ;)!

Ensure to have different key in the grouping vars

```r
library(tidyverse)
library(highcharter)

tm <- data.frame(level_1 = c("Unknown", "A", "A", "B", "B"),
                 level_2 = paste0(c("Unknown", "C", "C", "D", "D"), "2"),
                 value = abs(rnorm(5))) 

tm %>% hctreemap2(group_vars = c("level_1", "level_2"), size_var = "value")
```
